internal/cli/profile.go

```diff
@@ -182,3 +184,13 @@ func (p Profile) SetSession(accessToken, refreshToken string) {
 	p.SetString(keyAccessToken, accessToken)
 	p.SetString(keyRefreshToken, refreshToken)
 }
+
+// GetTelemetryMode gets the Telemetry Mode
+func (p Profile) GetTelemetryMode() telemetry.Mode {
+	return telemetry.NewMode(p.GetString(keyTelemetryMode))
```

Comment:
we won't need the `NewMode` (and subsequent validation logic) here, just simply wrap `p.GetString(keyTelemetryMode)` with `telemetry.Mode( ... )`

We won't ever save an invalid telemetry mode b/c the flag that gets parsed uses `Set` on `telemetry.Mode`

Reply:
why wouldn't we want to strengthen our guarantees here?  `telemetry.Mode( ... )` needlessly reduces orthogonality between profile and telemetry

Reply:
mostly because its simpler.  it's not a situation that could or should arise based on how the whole system is designed.  another reason is for future-proof, there's no reason today that `Mode` should ever be "constructed" in code (it's passed in through flags), so we shouldn't export anything that allows developers to do so down the line

Reply:
It's literally being constructed in code right here.  However instead of using a public interface with telemetry, you want us to depend on the internal representation of Mode as a string.  Also it's possible for a user to edit the config file, so it's not a super strong guarantee that it will always be a valid Mode string.  If you still want me to do a cast I will, I just want to give one last argument for why I disagree with this style of code.